lineman/app/components/thread_page/thread_page_controller.coffee

```diff
@@ -1,24 +1,43 @@
 angular.module('loomioApp').controller 'ThreadPageController', ($scope, $routeParams, $location, $rootScope, Records, MessageChannelService, CurrentUser, DiscussionFormService, ScrollService) ->
   $rootScope.$broadcast('currentComponent', { page: 'threadPage'})
 
-  $scope.$on 'threadPageEventsLoaded',    (event, sequenceId) =>
-    ScrollService.scrollTo("#sequence-#{sequenceId}") if @focusMode == 'activity'
-  $scope.$on 'threadPageProposalsLoaded', (event, proposalId) =>
-    ScrollService.scrollTo("#proposal-#{proposalId}") if @focusMode == 'proposal'
-
-  Records.discussions.findOrFetchByKey($routeParams.key).then (discussion) =>
-    @discussion = discussion
-    $rootScope.$broadcast('setTitle', @discussion.title)
-    @group = @discussion.group()
-    $rootScope.$broadcast('viewingThread', @discussion)
-    MessageChannelService.subscribeTo("/discussion-#{@discussion.key}", @onMessageReceived)
-    @setFocusMode()
-    ScrollService.scrollTo('.thread-context') if @focusMode == 'context'
-  , (error) ->
+  @performScroll = ->
+    focusedElement = @focusedElement()
+    if focusedElement && !@scrolledAlready
+      ScrollService.scrollTo focusedElement
+      @scrolledAlready = true
+
+  @focusedElement = ->
+    if $location.hash().match(/^proposal-\d+$/) and Records.proposals.find(@focusedProposalId)
+      "#proposal-#{@focusedProposalId}"
+    else if @discussion.lastSequenceId == 0 or @discussion.reader().lastReadSequenceId == -1
+      ".thread-context"
+    else if Records.events.findByDiscussionAndSequenceId(@discussion, @focusedSequenceId)
```

Comment:
Yea.. again can we focus on the new activity indicator? I think that would be real nice.

Is it easy to explain/tell me why you check for a record here?